sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/implementation/TableUtils.java

```diff
@@ -97,6 +97,32 @@ public static Throwable mapThrowableToTableServiceException(Throwable throwable)
         }
     }
 
+    /**
+     * Map a {@link Throwable} to {@link IllegalArgumentException} if the endpoint
+     * contains a table name, else it returns the original throwable
+     * @param throwable The throwable to map.
+     * @param tableName The name of the Table intended to be operated on
+     * @param endpoint The endpoint of the {@link com.azure.data.tables.TableServiceClient},
+     *                 {@link com.azure.data.tables.TableServiceAsyncClient}
+     * @return A Throwable that is either an instance of {@link IllegalArgumentException} or the original throwable.
+     */
+    public static Throwable mapThrowableToIllegalArgumentExceptionWhenTableNamePresent(Throwable throwable,
+                                                                                       String tableName,
+                                                                                       String endpoint) {
+        /*if (endpoint.contains(tableName)) {
+            return new IllegalArgumentException(
+                "Table name found within client endpoint URI. Remove table name from endpoint.");
+        }
+        return throwable;*/
+        String adjustedEndpoint = (endpoint.endsWith("/")) ? endpoint.substring(0, endpoint.length() - 1) : endpoint;
+        if (adjustedEndpoint.endsWith(tableName)) {
```

Comment:
You can technically create a table simply named "net", meaning that if we have the correct endpoint "https://myaccount.table.core.windows.net" given to our builder and a create table call fails under any circumstances, we could be providing the wrong error message to a user. We should instead grab the URI's path and check if the table name is present there. There are other checks we could do but I believe this should suffice to avoid false positives.